### ISSUE TITLE
Fatal error in TrailingArrayCommaSniff::process with PHP_CodeSniffer 4.0.0

### ISSUE THREAD

**Issue by zener05** [slevomat/coding-standard#1803]
## Bug Description

  When using `slevomat/coding-standard` version 8.24.0 with `squizlabs/php_codesniffer` 4.0.0, a fatal error occurs:

  PHP Fatal error:  Declaration of SlevomatCodingStandard\Sniffs\Arrays\TrailingArrayCommaSniff::process(PHP_CodeSniffer\Files\File $phpcsFile, int
  $stackPointer): void must be compatible with PHP_CodeSniffer\Sniffs\Sniff::process(PHP_CodeSniffer\Files\File $phpcsFile, $stackPtr) in
  /vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/Arrays/TrailingArrayCommaSniff.php on line 31

  ## Environment

  - `slevomat/coding-standard`: 8.24.0
  - `squizlabs/php_codesniffer`: 4.0.0
  - PHP: 8.3.23

  ## Steps to Reproduce

  1. Install dependencies:
     ```bash
     composer require --dev slevomat/coding-standard:^8.16 squizlabs/php_codesniffer:^4.0
     composer update

  2. Run ECS or phpcs with Slevomat sniffs
  3. Fatal error occurs

## Workaround

  Downgrading to compatible versions resolves the issue:
```json
{
    "require-dev": {
      "slevomat/coding-standard": "^8.16 <8.22",
      "squizlabs/php_codesniffer": "^3.11"
    }
}
```

## Expected Behavior

  The sniff should work with PHP_CodeSniffer 4.0.0, as declared in composer.json requirements ("squizlabs/php_codesniffer": "^4.0.0").

## Actual Behavior

  Fatal error due to method signature incompatibility between TrailingArrayCommaSniff::process() and the interface Sniff::process().
